File: app/src/main/java/android/academy/nytimes/data/NewsItem.java
package android.academy.nytimes.data;

import android.support.annotation.Nullable;

import java.util.Date;

public class NewsItem {

    private final String title;
    @Nullable
    private final String imageUrl;
    private final String category;
    private final Date publishDate;
    private final String previewText;
    private final String url;

    public NewsItem(String title, @Nullable String imageUrl, String category, Date publishDate, String previewText, String url) {
        this.title = title;
        this.imageUrl = imageUrl;
        this.category = category;
        this.publishDate = publishDate;
        this.previewText = previewText;
        this.url = url;
    }

    public String getTitle() {
        return title;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public String getCategory() {
        return category;
    }

    public Date getPublishDate() {
        return publishDate;
    }

    public String getPreviewText() {
        return previewText;
    }

    public String getUrl() {
        return url;
    }

}

File: app/src/main/java/android/academy/nytimes/data/NewsItem.kt
package android.academy.nytimes.data

import java.util.*

sealed class NewsItem

data class CategorizedNewsItem(val title: String,
                               val imageUrl: String?,
                               val category: String,
                               val publishDate: Date,
                               val previewText: String,
                               val url: String) : NewsItem()

data class UncategorizedNewsItem(val title: String,
                                 val imageUrl: String?,
                                 val publishDate: Date,
                                 val previewText: String,
                                 val url: String) : NewsItem()
